### ISSUE TITLE
Missing styles after 12 -> 13 upgrade

### ISSUE THREAD

**Issue by lnicola** [nextcloud/server#8316]
### Actual behaviour
I have no styles after logging in:

![image](https://user-images.githubusercontent.com/308347/36108453-92c342a2-1025-11e8-9ec8-c4b288b6b95f.png)

The browser is requesting e.g. `/css/core/9819c5b1-server.css?v=38191880-0`, while the proper names are `/css/core/50273b4f7cebd6b8f92d7809b8084497-server.css`. If I load the latter addres in the browser, it works.

I don't know where the `9819c5b1` hash comes from. `occ maintenance:repair`, `occ maintenance:theme:update` and `occ files:cleanup` had no effect.

### Server configuration

**Operating system**: Arch Linux

**Web server:** nginx 1.13.7

**Database:** PostgrSQL 10.1

**PHP version:** 7.2.2

**Nextcloud version:** 13.0.0

**Updated from an older Nextcloud/ownCloud or fresh install:** Upgraded from Nextcloud 12

**Where did you install Nextcloud from:** Distribution package

**Signing status:**
<details>
<summary>Signing status</summary>

```
No errors have been found.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
Enabled:
  - activity: 2.6.1
  - dav: 1.4.6
  - federatedfilesharing: 1.3.1
  - files: 1.8.0
  - files_external: 1.4.1
  - files_pdfviewer: 1.2.0
  - files_sharing: 1.5.0
  - files_texteditor: 2.5.1
  - files_trashbin: 1.3.0
  - files_versions: 1.6.0
  - files_videoplayer: 1.2.0
  - firstrunwizard: 2.2.1
  - gallery: 18.0.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.1.0
  - music: 0.5.5
  - nextcloud_announcements: 1.2.0
  - notifications: 2.1.2
  - oauth2: 1.1.0
  - password_policy: 1.3.0
  - provisioning_api: 1.3.0
  - serverinfo: 1.3.0
  - sharebymail: 1.3.0
  - theming: 1.4.1
  - twofactor_backupcodes: 1.2.3
  - updatenotification: 1.3.0
  - workflowengine: 1.3.0
Disabled:
  - admin_audit
  - audioplayer
  - comments
  - encryption
  - federation
  - survey_client
  - systemtags
  - user_external
  - user_ldap
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "nextcloud.dend.ro"
        ],
        "memcache.local": "\\OC\\Memcache\\Redis",
        "filelocking.enabled": true,
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0
        },
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "pgsql",
        "version": "13.0.0.14",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "UTC",
        "installed": true,
        "mail_smtpmode": "smtp",
        "mail_smtpsecure": "ssl",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauthtype": "PLAIN",
        "mail_smtpauth": 1,
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "theme": "",
        "loglevel": 2,
        "maintenance": false,
        "updater.release.channel": "stable"
    }
}
```
</details>

**Are you using external storage, if yes which one:** Local

**Are you using encryption:** No

**Are you using an external user-backend, if yes which one:** No

### Client configuration
**Browser:** Firefox 60

**Operating system:** Linux x64

#### Web server error log
<details>
<summary>Web server error log</summary>

```
N/A
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
{"reqId":"D7jLsdgsh98jOHrELSWg","level":3,"time":"2018-02-12T16:36:12+00:00","remoteAddr":"***REMOVED SENSITIVE VALUE***","user":"***REMOVED SENSITIVE VALUE***","app":"PHP","method":"GET","url":"\/css\/core\/9819c5b1-server.css?v=38191880-0","message":"file_get_contents(\/usr\/share\/webapps\/nextcloud\/data\/appdata_ocmhf35boin0\/css\/core\/9819c5b1-server.css.gzip): failed to open stream: No such file or directory at \/usr\/share\/webapps\/nextcloud\/lib\/private\/Files\/Storage\/Local.php#214","userAgent":"Mozilla\/5.0 (X11; Linux x86_64; rv:60.0) Gecko\/20100101 Firefox\/60.0","version":"13.0.0.14"}
{"reqId":"h7hBYhWBDP2O5zNlRuLM","level":3,"time":"2018-02-12T16:37:24+00:00","remoteAddr":"***REMOVED SENSITIVE VALUE***","user":"***REMOVED SENSITIVE VALUE***","app":"PHP","method":"GET","url":"\/apps\/files\/","message":"Invalid argument supplied for foreach() at \/usr\/share\/webapps\/nextcloud\/lib\/private\/Template\/SCSSCacher.php#151","userAgent":"Mozilla\/5.0 (X11; Linux x86_64; rv:60.0) Gecko\/20100101 Firefox\/60.0","version":"13.0.0.14"}
{"reqId":"BhrUg6VXROjPGNKiWlhL","level":3,"time":"2018-02-12T16:42:45+00:00","remoteAddr":"***REMOVED SENSITIVE VALUE***","user":"***REMOVED SENSITIVE VALUE***","app":"PHP","method":"GET","url":"\/ocs\/v2.php\/apps\/notifications\/api\/v2\/notifications","message":"file_get_contents(\/usr\/share\/webapps\/nextcloud\/data\/appdata_ocmhf35boin0\/avatar\/***REMOVED SENSITIVE VALUE***\/avatar.png): failed to open stream: No such file or directory at \/usr\/share\/webapps\/nextcloud\/lib\/private\/Files\/Storage\/Local.php#214","userAgent":"Mozilla\/5.0 (X11; Linux x86_64; rv:60.0) Gecko\/20100101 Firefox\/60.0","version":"13.0.0.14"}
[and a lot of other similar errors]
```
</details>

#### Browser log
<details>
<summary>Browser log</summary>

```
Some requests like /css/core/9819c5b1-server.css?v=38191880-0 return 200, but empty responses.
```
</details>

**Comment by gurussia** [comment]
php 7.2.2 how to get it work. I activate php7.2 and now how you see on my photos in index.php, after you login in nextcloud it comment white page, everything is working, only file application is not working. WIth php7.2 everything works very fast. Why nextcloud don;t make support for php7.2 and why nextcloud developers don't make repair tools in case of white screens  i try more than 10 times occ maintenance:repair or etc. and it didn't help.

**Comment by lnicola** [comment]
@gurussia You might be having a different problem, and that will get more attention if you report it as a new issue instead of here. You'll need to check the server logs, maybe there are some clues there. No software is perfect, and no programmer will think ahead about every possible thing that might go wrong, in every setup.

PS: You can paste images from clipboard on GitHub, just press `Ctrl-V` (or `Cmd-V`).

**Comment by qq690712575** [comment]
![qq 20180303194703](https://user-images.githubusercontent.com/17699491/36933940-6e505166-1f1c-11e8-84da-b9bc9dd5e68d.png)

**Comment by tflidd** [comment]
Did you try to delete the server cache, there is a known bug (fix will be in NC 13.0.1):
https://github.com/nextcloud/server/issues/8043

**Comment by lnicola** [comment]
Removing the `css` directory like described [here](https://github.com/nextcloud/server/issues/8043#issuecomment-360776226) fixed the issue for me. I also removed `js`, just in case.
